cli/cli-client/src/main/java/org/infinispan/cli/commands/User.java

```diff
@@ -0,0 +1,350 @@
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName = 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
+         userTool.removeUser(username);
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Password.CMD, description = "Changes a user's password")
+   public static class Password extends CliCommand {
+      public static final String CMD = "password";
+
+      @Argument(description = "The username for the user", required = true)
+      String username;
+
+      @Option(description = "The password for the user", shortName = 'p')
+      String password;
+
+      @Option(description = "The realm ", defaultValue = UserTool.DEFAULT_REALM_NAME, shortName = 'r')
+      String realm;
+
+      @OptionList(description = "The algorithms used to encrypt the password", shortName = 'a', completer = EncryptionAlgorithmCompleter.class)
+      List<String> algorithms;
+
+      @Option(description = "Whether the password should be stored in plain text", name = "plain-text", hasValue = false)
+      boolean plainText;
+
+      @Option(description = "The path of the users.properties file", name = "users-file", shortName = 'f')
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName = 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) {
+         if (password == null) { // Get the password interactively
+            try {
+               password = getPasswordInteractively(invocation);
+            } catch (InterruptedException e) {
+               return CommandResult.FAILURE;
+            }
+         }
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
+         userTool.modifyUser(username, password, realm, UserTool.Encryption.valueOf(plainText), null, algorithms);
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Groups.CMD, description = "Sets a user's groups")
```

Comment:
probably a limitation of aesh, but you can't use `user groups -g <group> <user>`
```
[disconnected]> user groups -g abc pedro
The username for the user is required for this command.
[disconnected]> user groups pedro -g abc
```

Reply:
That looks like a bug